### ISSUE TITLE
Research possibility to support multiple javascript engines

### ISSUE THREAD

**Issue by srs** [purplejs/purplejs#58]
Research possibility to support multiple javascript engines. If we make the Java/JS/Java bridge better we could support both Nashorn and J2V8 engines.

**Comment by sergehuber** [comment]
This is possible through the Java Scripting API by clearly identifying the engine used.

**Comment by srs** [comment]
Yes, I know. But the PurpleJS implementation and some API's is tied to Nashorn right now. Also the require implementation and the special Java-JS bridge is also Nashorn specific.

**Comment by sergehuber** [comment]
Ok thanks for the details. Indeed this might make it more difficult to bridge to other JS implementation
